src/spaces/continuous.jl

```diff
@@ -410,3 +410,230 @@ function Base.iterate(iter::PairIterator, i = 1)
     id1, id2 = p
     return (iter.agents[id1], iter.agents[id2]), i + 1
 end
+
+#######################################################################################
+# FMP Algorithm
+#######################################################################################
+export fmp_update_vel, fmp_update_interacting_pairs, FMP_Parameters, fmp_parameter_init
+
+"""
+    FMP_Parameters
+The parameters for the FMP model as defined in the FMP paper. Helper function `FMP_Parameter_Init`
+is used to initialize the struct with default parameters. Users can modify FMP default parameters
+through the usage of keyword arguments. See [the original paper](https://arxiv.org/abs/1909.05415)
+for a complete description of model parameters. Parameters included in this struct that are not 
+present in the original paper are:
+- `obstacle_list`: list of obstacles in the state space
+- `interaction_array`: n x n boolean array (n = number of agents) where `interaction_array[i,j]=1`
+indicates that the ith and jth agents are interacting. Used so that `interacting_pairs` is only called
+once per iteration rather than every time `update_vel` is called.
+- `agents`: an iterator of the interacting agents from `interacting_pairs` at each time step.
```

Comment:
This documentation string needs improvement. Why don't you state all parameters that this struct takes in? I don't think it is useful to refer the user to some paper, when we can simply state this information here.

Reply:
That is fair - I will update it to describe the parameters in the doc string. I will note that some of the parameters (`c1`, `c2`) seem to be empirically defined hyperparameters. The paper literally just refers to them as "two positive constant values".

In general my thought was to provide `fmp_parameter_init()` to remove any user guessing about how to parameterize the FMP algorithm by providing the default values used in the original paper. I want to expose the underlying parameters in case users decide they want to modify them for some reason but wanted to make the algorithm as plug and play as possible. My approach is admittedly a little clunky so if you have a better idea of how to approach this I am interested in hearing it.